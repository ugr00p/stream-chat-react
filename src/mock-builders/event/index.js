export { default as dispatchMessageNewEvent } from './messageNew';
export { default as dispatchChannelUpdatedEvent } from './channelUpdated';
export { default as dispatchChannelDeletedEvent } from './channelDeleted';
export { default as dispatchChannelTruncatedEvent } from './channelTruncated';
export { default as dispatchNotificationAddedToChannelEvent } from './notificationAddedToChannel';
export { default as dispatchNotificationMessageNewEvent } from './notificationMessageNew';
export { default as dispatchNotificationRemovedFromChannel } from './notificationRemovedFromChannel';
export { default as dispatchConnectionRecoveredEvent } from './connectionRecovered';
export { default as dispatchNotificationMutesUpdated } from './notificationMutesUpdated';
