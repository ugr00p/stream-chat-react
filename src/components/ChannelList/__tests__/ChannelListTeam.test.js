/* eslint-disable sonarjs/no-unused-collection */
import React from 'react';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { getTestClientWithUser } from 'mock-builders';
import { ChannelListTeam } from '..';

// Wierd hack to avoid big warnings
// Maybe better to find a better solution for it.
console.warn = () => null;

jest.mock('axios');

const Component = ({ client, loading = false, error = false }) => {
  return (
    <ChannelListTeam
      client={client}
      loading={loading}
      error={error}
      LoadingIndicator={() => <div>Loading Indicator</div>}
      LoadingErrorIndicator={() => <div>Loading Error Indicator</div>}
    >
      <div>children 1</div>
      <div>children 2</div>
    </ChannelListTeam>
  );
};

describe('ChannelListTeam', () => {
  afterEach(cleanup);

  let chatClientVishal;

  beforeEach(async () => {
    chatClientVishal = await getTestClientWithUser({ id: 'vishal' });
  });
  it('by default, should render sidebar, userbar and children', () => {
    const tree = renderer
      .create(<Component client={chatClientVishal} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('when `error` prop is true, `LoadingErrorIndicator` should be rendered', () => {
    const tree = renderer
      .create(<Component client={chatClientVishal} error />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('when `loading` prop is true, `LoadingIndicator` should be rendered', () => {
    const tree = renderer
      .create(<Component client={chatClientVishal} loading />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
