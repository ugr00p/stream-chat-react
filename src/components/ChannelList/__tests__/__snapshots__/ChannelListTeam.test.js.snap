// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelListTeam by default, should render sidebar, userbar and children 1`] = `
<div
  className="str-chat__channel-list-team"
>
  <div
    className="str-chat__channel-list-team__main"
  >
    <div
      className="str-chat__channel-list-team__header"
    >
      <div
        className="str-chat__channel-list-team__header--left"
      >
        <div
          className="str-chat__avatar str-chat__avatar--circle"
          data-testid="avatar"
          onClick={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "flexBasis": 40,
              "fontSize": 20,
              "height": 40,
              "lineHeight": "40px",
              "width": 40,
            }
          }
          title="vishal"
        >
          <div
            className="str-chat__avatar-fallback"
            data-testid="avatar-fallback"
          >
            v
          </div>
        </div>
      </div>
      <div
        className="str-chat__channel-list-team__header--middle"
      >
        <div
          className="str-chat__channel-list-team__header--title"
        >
          vishal
        </div>
        <div
          className="str-chat__channel-list-team__header--status undefined"
        />
      </div>
      <div
        className="str-chat__channel-list-team__header--right"
      >
        <button
          className="str-chat__channel-list-team__header--button"
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </button>
      </div>
    </div>
    <div>
      children 1
    </div>
    <div>
      children 2
    </div>
  </div>
</div>
`;

exports[`ChannelListTeam when \`error\` prop is true, \`LoadingErrorIndicator\` should be rendered 1`] = `
<div>
  Loading Error Indicator
</div>
`;

exports[`ChannelListTeam when \`loading\` prop is true, \`LoadingIndicator\` should be rendered 1`] = `
<div>
  Loading Indicator
</div>
`;
